auth/src/main/java/feast/auth/authorization/HttpAuthorizationProvider.java

```diff
@@ -0,0 +1,141 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.auth.authorization;
+
+import feast.auth.generated.client.api.DefaultApi;
+import feast.auth.generated.client.invoker.ApiClient;
+import feast.auth.generated.client.invoker.ApiException;
+import feast.auth.generated.client.model.CheckAccessRequest;
+import java.util.Map;
+import org.hibernate.validator.internal.constraintvalidators.bv.EmailValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+/**
+ * HTTPAuthorizationProvider uses an external HTTP service for authorizing requests. Please see
+ * auth/src/main/resources/api.yaml for the API specification of this external service.
+ */
+public class HttpAuthorizationProvider implements AuthorizationProvider {
+
+  private static final Logger log = LoggerFactory.getLogger(HttpAuthorizationProvider.class);
+
+  private final DefaultApi defaultApiClient;
+
+  /**
+   * The default subject claim is the key within the Authentication object where the user's identity
+   * can be found
+   */
+  private final String DEFAULT_SUBJECT_CLAIM = "email";
+
+  /**
+   * Initializes the HTTPAuthorizationProvider
+   *
+   * @param options String K/V pair of options to initialize the provider with. Expects at least a
+   *     "basePath" for the provider URL
+   */
+  public HttpAuthorizationProvider(Map<String, String> options) {
+    if (options == null) {
+      throw new IllegalArgumentException(
+          "Cannot pass empty or null options to HTTPAuthorizationProvider");
+    }
+
+    ApiClient apiClient = new ApiClient();
+    apiClient.setBasePath(options.get("authorizationUrl"));
+    this.defaultApiClient = new DefaultApi(apiClient);
+  }
+
+  /**
+   * Validates whether a user has access to a project
+   *
+   * @param projectId Name of the Feast project
+   * @param authentication Spring Security Authentication object
+   * @return AuthorizationResult result of authorization query
+   */
+  public AuthorizationResult checkAccessToProject(String projectId, Authentication authentication) {
+
+    CheckAccessRequest checkAccessRequest = new CheckAccessRequest();
+    Object context = getContext(authentication);
+    String subject = getSubjectFromAuth(authentication, DEFAULT_SUBJECT_CLAIM);
+    checkAccessRequest.setAction("ALL");
+    checkAccessRequest.setContext(context);
+    checkAccessRequest.setResource(projectId);
```

Comment:
we probably need to change this to include the resourcetype like `org.feast.project:{projectId}` or something

Reply:
Oops, I fixed it but havent pushed yet.

Reply:
I had the do-not-merge on for protection ;)

But no worries, this at least allows your team to move ahead. I'll submit a patch tomorrow. I am going to bed now.